lib/ansible/modules/system/hostname.py

```diff
@@ -724,7 +738,8 @@ class NeonHostname(Hostname):
 def main():
     module = AnsibleModule(
         argument_spec=dict(
-            name=dict(required=True)
+            name=dict(type='str', required=True),
+            use=dict(type='str', choices=['generic', 'debian', 'sles', 'redhat', 'alpine', 'systemd', 'openrc', 'openbsd', 'solaris', 'freebsd'])
```

Comment:
Could you do something like `choices=STRATS.keys()` here to avoid duplicating the list?